### DISCUSSION TITLE
Langroid with Huggingface OpenAI-compatible model

### DISCUSSION THREAD

**Discussion by jduke99** [langroid/langroid#596] [Q&A]
Hi there, 

I have a Huggingface model deployed at: http://my_local_server:9001/v1 . It is deployed via TGI.

I can CURL it at http://my_local_server:9001/v1/completions

The model is "microsoft/Phi-3.5-mini-instruct"

I cannot seem to configure my chat definition to access this LLM correctly.  Any help?

**Comment by pchalasani** [comment]
@jduke99 I expect it should work if you do this:

```python
import langroid.language_models as lm
llm = lm.OpenAIGPT(
    config = lm.OpenAIGPTConfig(chat_model="local/my_local_server:9001/v1")
)
llm.chat("3+4=?")
```
Or instead of `my_local_server`, use the actual numeric IP (digits and dots only, omit the `http://`  part).

Docs here for reference: https://langroid.github.io/langroid/tutorials/local-llm-setup/#harder-with-oobabooga
(it is about ooba, but other local-serving scenarios work similarly)
